### PR TITLE
Fixed an issue where Universal Render Pipeline with disabled antiAliasing was overwriting QualitySettings.asset

### PR DESCRIPTION
**Pull request by lukaschod**
# **Please read;**
## **PR Workflow for the Graphics repository:**
* **All PRs must be opened as draft initially**
* Reviewers can be added while the PR is still in draft
* The PR can be marked as “Ready for Review” once the reviewers have confirmed that **no more changes are needed**
* Tests will start automatically after the PR is marked as “Ready for Review”
* **Do not use [skip ci]** - this can break some of our tooling
* Read the [Graphics repository & Yamato FAQ](http://go/graphics-yamato-faq).

### Checklist for PR maker
- [x] Have you added a backport label (if needed)? For example, the `need-backport-*` label. After you backport the PR, the label changes to `backported-*`.
- [x] Have you updated the changelog? Each package has a `CHANGELOG.md` file.
- [ ] Have you updated or added the documentation for your PR? When you add a new feature, change a property name, or change the behavior of a feature, it's best practice to include related documentation changes in the same PR. If you do add documentation, make sure to add the relevant Graphics Docs team member as a reviewer of the PR. If you are not sure which person to add, see the [Docs team contacts sheet](https://docs.google.com/spreadsheets/d/1rgUWWgwLFEHIQ3Rz-LnK6PAKmbM49DZZ9al4hvnztOo/edit#gid=1058860420).
- [ ] Have you added a graphic test for your PR (if needed)? When you add a new feature, or discover a bug that tests don't cover, please add a graphic test.

---
### Purpose of this PR
Fixed an issue where Universal Render Pipeline with disabled antiAliasing was overwriting QualitySettings.asset on frequent cases (https://fogbugz.unity3d.com/f/cases/1219159/).

---
### Testing status
- [x] Tested that issue no longer reproduces in case.
- [x] Checked the code if XR needs fixing. Apperently they are handling it in native side 
```
void XRDisplaySubsystem::SetMSAALevel(int level)
{
    level = (level > 0) ? level : 1;
    m_TextureManager.SetMSAALevel(level);
}
```
- [x] Test if QualitySettings.antiAliasing is correctly update with URP.

---
### Comments to reviewers
I think we should discuss a bit more about antiAliasing value inconsistency between builtin and SRP. I have feeling it might bite us in the future again.

### CODE REVIEW

**Review by phi-lira — APPROVED** [review]

File: com.unity.render-pipelines.universal/Runtime/UniversalRenderPipeline.cs [hunk 1]
```diff
@@ -144,8 +144,12 @@ public UniversalRenderPipeline(UniversalRenderPipelineAsset asset)
...
 
+            // In QualitySettings.antiAliasing disabled state uses value 0, where in URP 1
+            int qualitySettingsMsaaSampleCount = QualitySettings.antiAliasing > 0 ? QualitySettings.antiAliasing : 1;
+            bool msaaSampleCountNeedsUpdate = qualitySettingsMsaaSampleCount != asset.msaaSampleCount;
+
             // Let engine know we have MSAA on for cases where we support MSAA backbuffer
-            if (QualitySettings.antiAliasing != asset.msaaSampleCount)
+            if (msaaSampleCountNeedsUpdate)
             {
                 QualitySettings.antiAliasing = asset.msaaSampleCount;
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Runtime/UniversalRenderPipeline.cs:154]: don't we need to convert the data when setting antialiasing as well?

Comment (phi-lira) [reply com.unity.render-pipelines.universal/Runtime/UniversalRenderPipeline.cs:154]: @thomas-zeng I see @lukaschod already checked SetMSAALevel but pinging you here for awareness

**Review by ellioman — APPROVED** [review]

File: com.unity.render-pipelines.universal/Runtime/UniversalRenderPipeline.cs (hunk 1 above)
Comment (lukaschod) [reply com.unity.render-pipelines.universal/Runtime/UniversalRenderPipeline.cs:154]: conversion happens inside native side

**Comment by ellioman** [comment]
The failures on Yamato are known and not caused by this PR.

**Pull request merged**